library/src/updater.rs

```diff
@@ -103,7 +115,11 @@ fn libapp_path_from_settings(original_libapp_paths: &[String]) -> Result<PathBuf
 /// The yaml string is the contents of the `shorebird.yaml` file.
 /// The `AppConfig` struct is information about the running app and where
 /// the updater should keep its cache.
-pub fn init(app_config: AppConfig, yaml: &str) -> Result<(), UpdateError> {
+pub fn init(
+    app_config: AppConfig,
+    file_provider: Box<dyn ExternalFileProvider>,
```

Comment:
I do think you're going to end up with this being Arc in a later patch, since that makes the lifetime more clear.